### ISSUE TITLE
can not check protected-member in member function

### ISSUE THREAD

**Issue by Bill-Tao-Yang** [PyCQA/pylint#3949]
### Steps to reproduce
pylint the following

```
class Test():
    def __init__(self):
        self._foo._bar = True
```

### Current behavior
Nothing.
```
-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.66/10, +0.34)
```

### Expected behavior
when I access to protected members. It should be warn `W0212: Access to a protected member _bar of a client class (protected-access)` 

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.7.4 (default, Sep  2 2019, 20:44:09)
[GCC 5.4.0 20160609]
```

**Comment by hippo91** [comment]
@Bill-Tao-Yang thanks for your report.
By default `pylint` doesn't emit such message for access to protected members inside special methods. The reason is that such access could be legitimate. For example when implementing `__eq__(self, other)` you may want to compare protected member of `other` and those of the current instance.  
However since #3864 you can enable such check by adding `check-protected-access-in-special-methods` in your config file.
